src/dirac_cwl_proto/execution_hooks/requirement_validator.py

```diff
@@ -0,0 +1,217 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar
+
+from cwl_utils.parser.cwl_v1_2 import CommandLineTool, Workflow, WorkflowStep
+from pydantic import BaseModel, ConfigDict
+
+
+class RequirementValidator(BaseModel, ABC):
```

Comment:
Could you explain the purpose of `RequirementValidator` please?
I am not sure to understand why we would need to have this abstract validator.

Do you foresee a need to validate other CWL requirements?
I have the impression that you are thinking ahead and try to abstract your work. It is valuable but may be too premature in this case:
- you have a single inherited class
- I am not entirely sure we want more than that (but may be you have specific ideas in mind)